content/js1/sprints/3/prep/index.md

```diff
@@ -8,4 +8,634 @@ backlog= 'Module-JS1'
+Write a program that defines a variable `eleniAge`.
+
+If Eleni is less than 18, the program should print out nothing to the console.
+
+If Eleni is 18 years old or older, the program should print out:
+
+```bash
+Eleni is old enough to drive
+```
+
+{{</note>}}
+
+{{<note type="exercise" title="Exercise 2.2">}}
+
+Modify the program above so that if Eleni is less than 18 the program prints out:
+
+```bash
+Eleni is not old enough to drive
+```
+
+{{</note>}}
+
+### 🧠 Applying new knowledge
+
+So for `formatAs12HourClock` we said part of the strategy for handling `"23:00"` would involve checking that the `hours` value is less than 12. For this purpose, we can use another comparison operator: `>`.
+`>` will check if the value on the left of the operator is less than the value on the right of the operator.
+So `3 > 12` would evaluate to be `false`, as `3` is not greater than `12`.
+
+So provided we have an expression for the hours from the time, we can write a conditional statement as follows:
+
+```js
+if (expressionsForHours < 12) {
+  // do code to format the 12 hours
+}
+```
+
+To complete the logic, we now have to achieve a sub-goal:
+
+> 🎯 Sub-goal: Find an expression for the hours from the `time` input
+
+### 🧶 Accessing strings
+
+Given a `time` string we need to access the _first 2 characters_ of the string which represent the hours.
+
+Strings are **zero-indexed**.
+**Index** means **position**, so **zero-indexed** means we start counting character positions from `0` onwards.
+
+Here are the positions/indexes for `"23:00"`
+
+|   index   |   0   |   1   | 2     | 3     | 4     |
+| :-------: | :---: | :---: | ----- | ----- | ----- |
+| character | `"2"` | `"3"` | `":"` | `"0"` | `"0"` |
+
+In JavaScript, we can use square bracket notation to access specific characters in the string using the index.
+
+```js
+time[0]; // evaluates to "2"
+time[1]; // evaluates to "3"
+time[2]; // evaluates to ":"
+// etc
+```
+
+However at the moment, square bracket access wil only give a single character.
+So we must use another method to extract _multiple_ characters from the given string.
+
+### ✂️ Extracting a slice
+
+We can use a function called `slice` to extract 1 or more characters from a string.
+
+[`slice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) is a function that can take 2 arguments: a start index and an end index. `slice` will return a section of the string from the start index up to but not including the start index.
+
+```js
+time; // holds the value "23:00"
+```
+
+```js
+time.slice(0, 2); // will access the characters below
+```
+
+|   index   |   0   |   1   |
+| :-------: | :---: | :---: |
+| character | `"2"` | `"3"` |
+
+So `time.slice(0,2)` will evaluate to `"23"` when the `time` is `"23:00"`. Finally we must convert `"23"` to the number `23`, otherwise we can't compare this value properly. We can use the `+` operator to convert strings to numbers by writing the following:
```

Comment:
I'd much rather introduce `parseInt` or the `Number()` constructor rather than `+` as a coercing operator... If someone introduced `+someString` in a PR I was reviewing, I'd ask them to change it out... WDYT?

Reply:
Yeah no problem - let's use `Number()`